### ISSUE TITLE
idea: allow wallwalking on buildables

### ISSUE THREAD

**Issue by bmorel** [Unvanquished/gameplay#20]
I think it could be interesting to allow wallwalking on buildables. See that dretch on your armory? Are you sure you want to fire that luci shot at it?

Another, and likely more interesting, side effect this would have is that dretches and mantises would no longer be forced to jump over buildables, which would, I think, make them more interesting to play.

**Comment by illwieckz** [comment]
Yes there is, done.

What we can't do is transfer issue across organizations (we can transfer repositories across organizations but not issues), meaning for example we can't move from `Unvanquished/Unvanquished` to `UnvanquishedAssets/unvanquished_src.dpkdir`…

**Comment by bmorel** [comment]
Typically the kind of cases when I'm happier to be proven wrong than to be proven right. Thanks.

**Comment by illwieckz** [comment]
With Gitlab be we would be able to do sub-groups, so maybe we would be able to transfer among everything while keeping “Game” and “Assets” groups because that would be groups belonging to the same organization…

**Comment by bmorel** [comment]
maybe this can be closed, then?

**Comment by bmorel** [comment]
TL;DR: I agree that current collisions do not look nice to the eye at all but this is unrelated to this topic.

AABB have horrible collisions only if they do not match the model's shape closely enough (armory, turrets, every cylinder actually, and there's lot of those for buildables).
I notice that on (some) alien buildables, BBoxes are much, much closer to the model, even if this means putting minor parts ouf the model (barricade, egg (which even have a strange tendency to destroy stuff around when growing, dunno why), acid tubes). This should (yes, it think it should) probably be done for some human models (armory, telenode (less important, but still)).

There are probably other solutions doable (allow to use TT_CAPSULE or TT_BISPHERE instead of AABB; allow to have 1 external AABB filled with smaller boxes) but they all require more work and I do not have enough knowledge on the topic (nor time to spend on that for now, busy enough with bots).

I understand that the game's gameplay have to be limited by what engine can do, but I think discussing the problem of collisions (which each dev and player of unvanquished is well aware of existence, I'm certain of it) should not be in the gameplay bugtracker, but in unvanquished's one.

**Comment by bmorel** [comment]
> This should (yes, it think it should) probably be done for some human models (armory, telenode (less important, but still)).

Something I had forgotten when I wrote this is: original armory have a drawing which fits much better with the technical limitations of AABB.

As for Telenodes, the model is nice and adapted, but the BBox is just too big. Maybe it would be better to 1st try to have models which do not make the technical limitations obvious, and then implement this mechanism?
